CMakeLists.txt

```diff
@@ -191,387 +248,461 @@ target_link_libraries(helics_base INTERFACE Boostlibs::core)
-	endif()
+if(NOT WIN32)
+  if(HELICS_HAVE_ZEROMQ AND NOT ZMQ_USE_STATIC_LIBRARY)
+    get_target_property(zmqlibfile libzmq IMPORTED_LOCATION)
+    if(NOT zmqlibfile)
+      get_target_property(zmqlibfile libzmq IMPORTED_LOCATION_RELEASE)
+    endif()
+    # message(STATUS "zmqlib file =${zmqlibfile}")
+    get_filename_component(zmqdir ${zmqlibfile} DIRECTORY)
+    # message(STATUS "zmqdir path =${zmqdir}")
+    if(APPLE)
+      # TODO: Make sure it works for building ZMQ from source automatically
+      # using CMake
+      list(APPEND CMAKE_INSTALL_RPATH ${ZeroMQ_INSTALL_PATH}/lib)
+      list(APPEND CMAKE_BUILD_RPATH ${ZeroMQ_INSTALL_PATH}/lib)
+    endif()
+    list(APPEND CMAKE_INSTALL_RPATH "${zmqdir}")
+    list(APPEND CMAKE_BUILD_RPATH "${zmqdir}")
+  endif()
 endif()
 
-option(BUILD_APPS "Build helics applications to run various apps, including broker, player, recorder, and helics_apps" ON)
-
-FILE(GLOB KEY_LIBRARY_FILES  ${AUTOBUILD_INSTALL_PATH}/bin/*)
-#message(STATUS "key files ${KEY_LIBRARY_FILES}")
-
-option(DISABLE_LOGGING "disable all normal, debug, and trace logging in HELICS" OFF)
-
-cmake_dependent_option(DISABLE_TRACE_LOGGING "disable trace logging" OFF "NOT DISABLE_LOGGING" OFF)
-cmake_dependent_option(DISABLE_DEBUG_LOGGING "disable debug logging" OFF "NOT DISABLE_LOGGING" OFF)
+option(
+  BUILD_APPS
+  "Build helics applications to run various apps, including broker, player, recorder, and helics_apps"
+  ON)
+
+file(GLOB KEY_LIBRARY_FILES ${AUTOBUILD_INSTALL_PATH}/bin/*)
+# message(STATUS "key files ${KEY_LIBRARY_FILES}")
+
+option(DISABLE_LOGGING "disable all normal, debug, and trace logging in HELICS"
+       OFF)
+
+cmake_dependent_option(DISABLE_TRACE_LOGGING
+                       "disable trace logging"
+                       OFF
+                       "NOT DISABLE_LOGGING"
+                       OFF)
+cmake_dependent_option(DISABLE_DEBUG_LOGGING
+                       "disable debug logging"
+                       OFF
+                       "NOT DISABLE_LOGGING"
+                       OFF)
 mark_as_advanced(DISABLE_TRACE_LOGGING)
 mark_as_advanced(DISABLE_DEBUG_LOGGING)
 
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 # create the fmt header only targets
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 set(FMT_TEST OFF)
 set(FMT_ASSUME_CPP14_SUPPORT ON)
 set(FMT_SILENT ON)
-#get the FMT header only library
+# get the FMT header only library
 add_subdirectory(ThirdParty/fmt EXCLUDE_FROM_ALL)
-HIDE_VARIABLE(FMT_DOC)
-HIDE_VARIABLE(FMT_INSTALL)
-HIDE_VARIABLE(FMT_PEDANTIC)
-HIDE_VARIABLE(FMT_TEST)
-HIDE_VARIABLE(FMT_USE_CPP14)
+hide_variable(FMT_DOC)
+hide_variable(FMT_INSTALL)
+hide_variable(FMT_PEDANTIC)
+hide_variable(FMT_TEST)
+hide_variable(FMT_USE_CPP14)
```

Comment:
Does this work,  are macros case sensitive?

Reply:
I tested it and it built fine, no errors during the `make -j 8` process.